cloudevents/generic/baseclient.go

```diff
@@ -0,0 +1,216 @@
+package generic
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+
+	"open-cluster-management.io/api/cloudevents/generic/options"
+)
+
+const (
+	restartReceiverSignal = iota
+	stopReceiverSignal
+)
+
+type receiveFn func(ctx context.Context, evt cloudevents.Event)
+
+type baseClient struct {
```

Comment:
should we have a separated Connect(ctx context.Context) in baseClient? so we do not start goroutine during new?

Reply:
use connect method instead of new method